-1

**Failed Tests**:   Build 
**Size**: This PR adds an extra 72KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a21812/47594/summary.html
**COMMIT**: 2fab2d1e33dd618167179dcdb4153dfb91795f48
**CMSSW**: CMSSW_15_1_X_2025-08-07-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48696/47594/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @hjbossi cms-sw/cmssw#48597
 - @kk428 cms-sw/cmssw#48674
 - @see-saw28 cms-sw/cmssw#48539
 - @Dr15Jones cms-sw/cmssw#48688

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a21812/47594/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a21812/47594/git-merge-result


## Build

I found compilation error when building: 

<pre>>> Compiling  src/Mixing/Base/src/PileupRandomNumberGenerator.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_15_1_X_2025-08-07-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_1_X_2025-08-07-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_X_2025-08-07-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/cms/coral/CORAL_2_3_21-0fa00c0ec38eac2c3d528f5230f69e5c/include/LCG -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/pcre/8.43-2d141998cfe5424b8f7aff48035cc2da/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/boost/1.80.0-189b192d618e9605b04b60048d1376aa/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/clhep/2.4.7.1-d3a3e353d370e701238f7949a0d7909f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/curl/7.79.0-d8c0d5017cfad573d276c56bd08114b8/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/gsl/2.6-f7574c606b0ce57ff601d3ca9534cd01/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/lcg/root/6.32.13-e61674dd33920ceb725b332c4d0bf91b/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/tbb/v2022.0.0-79b5a917b0c13f831cd534a5b9f53a95/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/zlib/1.2.13-d217cdbdd8d586e845e05946de2796be/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/fmt/10.2.1-e35fd1db5eb3abc8ac0452e8ee427196/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/OpenBLAS/0.3.27-70a9dd2c9f309171934f13e3003b0540/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/tinyxml2/6.2.0-a0ad3950415fa3138d99b7da42eb4c9f/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/Mixing/Base/src/MixingBase/PileupRandomNumberGenerator.cc.d src/Mixing/Base/src/PileupRandomNumberGenerator.cc -o tmp/el8_amd64_gcc12/src/Mixing/Base/src/MixingBase/PileupRandomNumberGenerator.cc.o
>> Compiling  src/Mixing/Base/src/SecondaryEventProvider.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_15_1_X_2025-08-07-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_1_X_2025-08-07-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_X_2025-08-07-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/cms/coral/CORAL_2_3_21-0fa00c0ec38eac2c3d528f5230f69e5c/include/LCG -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/pcre/8.43-2d141998cfe5424b8f7aff48035cc2da/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/boost/1.80.0-189b192d618e9605b04b60048d1376aa/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/clhep/2.4.7.1-d3a3e353d370e701238f7949a0d7909f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/curl/7.79.0-d8c0d5017cfad573d276c56bd08114b8/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/gsl/2.6-f7574c606b0ce57ff601d3ca9534cd01/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/lcg/root/6.32.13-e61674dd33920ceb725b332c4d0bf91b/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/tbb/v2022.0.0-79b5a917b0c13f831cd534a5b9f53a95/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/zlib/1.2.13-d217cdbdd8d586e845e05946de2796be/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/fmt/10.2.1-e35fd1db5eb3abc8ac0452e8ee427196/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/OpenBLAS/0.3.27-70a9dd2c9f309171934f13e3003b0540/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02901/el8_amd64_gcc12/external/tinyxml2/6.2.0-a0ad3950415fa3138d99b7da42eb4c9f/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/Mixing/Base/src/MixingBase/SecondaryEventProvider.cc.d src/Mixing/Base/src/SecondaryEventProvider.cc -o tmp/el8_amd64_gcc12/src/Mixing/Base/src/MixingBase/SecondaryEventProvider.cc.o
In file included from src/Mixing/Base/src/PileupRandomNumberGenerator.cc:16:
src/Mixing/Base/src/PileupRandomNumberGenerator.h:57:17: error: 'edm::ConsumesCollector' has not been declared
   57 |   void consumes(edm::ConsumesCollector&& iC) const final;
      |                 ^~~
src/Mixing/Base/src/PileupRandomNumberGenerator.h:57:8: error: 'void PileupRandomNumberGenerator::consumes(int&&) const' marked 'final', but is not virtual
   57 |   void consumes(edm::ConsumesCollector&& iC) const final;
      |        ^~~~~~~~
</pre>

--- a/Mixing/Base/src/PileupRandomNumberGenerator.cc
+++ b/Mixing/Base/src/PileupRandomNumberGenerator.cc
@@ -67,7 +67,7 @@ std::vector<RandomEngineState> const& PileupRandomNumberGenerator::getLumiCache(
   return s_dummy;
 }
 
-void PileupRandomNumberGenerator::consumes(edm::ConsumesCollector&& iC) const {}
+edm::EDConsumerBase* PileupRandomNumberGenerator::consumer() { return nullptr; }
 
 void PileupRandomNumberGenerator::print(std::ostream& os) const {}
 

--- a/Mixing/Base/src/PileupRandomNumberGenerator.h
+++ b/Mixing/Base/src/PileupRandomNumberGenerator.h
@@ -54,7 +54,7 @@ private:
   std::vector<RandomEngineState> const& getEventCache(edm::StreamID const&) const final;
   std::vector<RandomEngineState> const& getLumiCache(edm::LuminosityBlockIndex const&) const final;
 
-  void consumes(edm::ConsumesCollector&& iC) const final;
+  edm::EDConsumerBase* consumer() final;
 
   /// For debugging purposes only.
   void print(std::ostream& os) const final;

